client/http-client/src/client.rs

```diff
@@ -225,10 +234,11 @@ impl ClientT for HttpClient {
 		.await
 	}
 
-	async fn batch_request<'a, R>(&self, batch: Vec<(&'a str, Option<ParamsSer<'a>>)>) -> Result<Vec<R>, Error>
+	async fn batch_request<'a, R>(&self, batch: BatchRequestBuilder<'a>) -> Result<Vec<R>, Error>
```

Comment:
yeah, this API is a bit different than the others i.e, takes the concrete type `BatchRequestBuilder` instead of `ToRpcParams` I think that makes sense as it supports both `positional params (array)` and `named params (map)` but not possible for folks to use there own impl of this.

Indeed it makes the API easier to understand and cleaner, cool